src/draw/lv_img_decoder.h

```diff
@@ -40,26 +40,44 @@ enum {
 
 typedef uint8_t lv_img_src_t;
 
+/**
+ * Base type for a decoder context.
+ * You'll likely bootstrap from this to make your own, if you need too
+ */
+typedef struct {
+    uint16_t    magic_id; /**!< Used to ensure a decoder context is not used in another decoder context*/
+} lv_base_dec_ctx_t;
```

Comment:
I'd rather do this:
```c
typedef struct {
    uint32_t    magic_id; /**!< Used to ensure a decoder context is not used in another decoder context*/
    void * user_data;
} lv_img_decoder_ctx_t;
```

All the specific stuff can be added in `user_data`. This `lv_img_decoder_ctx_t *` type could be used instead of `void *` in prototypes.

Reply:
Ok

Reply:
Okay. Could you imagine other typical data the can be added by default?

Reply:
A bit-flag/mask to tell what are the feature of the decoder: 
Vector, Animated, Backward (gif is not able to play backward), Cacheable (ffmpeg is not cachable since it can play a stream), etc...

That way, we can react in the EVENT_RESIZED if the flag is present to reset the decoder's state for a vector picture, for example.

Reply:
I like this high level approach. This way maybe `LV_RLOTTIE_CTRL_...` flags and play/pause functions could be part of `lv_img` and `lv_animbtn` could be a generic widget for any animate-able image source.

Reply:
Please check what I've come to. Now, there's a `caps` member based on bit combination of features.
I've updated all image decoder to fill the caps they support.
 
I've found some likely bug in lv_png's `decoder_info` for `SRC_VARIABLE` case, please confirm. 

Also, added a mode to draw perfectly for vector based format and resized object. Basically, I'm not using the resized event, but directly extract the information from the draw event.